### PR TITLE
Implement Difficulty Modes

### PR DESCRIPTION
**Pull request by roh518**
Implementation of the different difficulty modes, which largely just affects timers such as how often customers spawn.
Additionally, implement endless mode's gradual increase in spawn rate.

Initial scope expanded, and also implemented the ability to pause power ups for when pausing functionality is implemented, and reworked how the power ups are referred to in the code, now using an enum of power ups.

### CODE REVIEW

**Review by LouisWarren — CHANGES_REQUESTED** [review]
you broke my code and now i need to do it again

**Comment by LouisWarren** [comment]
that was a joke im not mad dont worry

**Review by LouisWarren — APPROVED** [review]
1 out of 3 done! well done Ross, looks good

File: core/src/cs/eng1/piazzapanic/chef/Chef.java [hunk 1]
```diff
@@ -130,7 +131,16 @@ public void draw(Batch batch, float parentAlpha) {
     @Override
     public void act(float delta) {
         Vector2 movement = getInput()
-            .scl(speed * (PlayerState.getInstance().getBuffActive(0) ? 2 : 1));
+            .scl(
+                speed *
+                (
+                    PlayerState
+                            .getInstance()
```
Comment (dislocated-su) [inline core/src/cs/eng1/piazzapanic/chef/Chef.java:138]: Gotta love the prettifier making totally not delusional changes

**Review by dislocated-su — APPROVED** [review]

**Pull request merged**